src/test/javascript/spec/component/exercises/shared/team-submission-sync.component.spec.ts

```diff
@@ -0,0 +1,105 @@
+                MockProvider(SessionStorageService),
+                MockProvider(JhiWebsocketService),
+                { provide: AccountService, useClass: MockAccountService },
+                { provide: TranslateService, useClass: MockTranslateService },
+                { provide: HttpClient, useClass: MockHttpService },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(TeamSubmissionSyncComponent);
+                component = fixture.componentInstance;
+
+                websocketService = TestBed.inject(JhiWebsocketService);
+
+                component.exerciseType = ExerciseType.TEXT;
+                const participation = new StudentParticipation(ParticipationType.STUDENT);
+                participation.id = 3;
+                component.participation = participation;
+                const textSubmissionWithParticipation = new TextSubmission();
+                textSubmissionWithParticipation.participation = new StudentParticipation();
+                submissionObservableWithParticipation = of(textSubmissionWithParticipation);
+                component.submissionObservable = submissionObservableWithParticipation;
+                currentUser = new User();
+                currentUser.login = 'ge12ebc';
+                component.currentUser = currentUser;
+
+                submissionSyncPayload = new SubmissionSyncPayload();
+                submissionSyncPayload.sender = currentUser;
+                const submission = new TextSubmission();
+                submission.id = 12;
+                submissionSyncPayload.submission = submission;
+            });
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    it('ngOnInit should work correctly', () => {
+        const expectedWebsocketTopic = '/topic/participations/3/team/text-submissions';
+        const submissionSyncObservable = of(submissionSyncPayload);
+
+        const receiveSubmissionEventEmitter = jest.spyOn(component.receiveSubmission, 'emit');
+        const websocketSubscribeSpy = jest.spyOn(websocketService, 'subscribe');
+        const websocketReceiveMock = jest.spyOn(websocketService, 'receive').mockReturnValue(submissionSyncObservable);
+        const websocketSendSpy = jest.spyOn(websocketService, 'send');
+
+        component.ngOnInit();
+
+        expect(component.websocketTopic).toBe(expectedWebsocketTopic);
+        expect(websocketSubscribeSpy).toHaveBeenCalledTimes(1);
+        expect(websocketSubscribeSpy).toHaveBeenCalledWith(expectedWebsocketTopic);
+
+        // checks for setupReceiver
+        expect(websocketReceiveMock).toHaveBeenCalledTimes(1);
+        expect(websocketReceiveMock).toHaveBeenCalledWith(expectedWebsocketTopic);
+        let submissionPayload: SubmissionSyncPayload | undefined;
+        submissionSyncObservable.subscribe((result: SubmissionSyncPayload) => (submissionPayload = result));
```

Comment:
It seemed to me getting `submissionPayload` by subscribing should give the same results with just having `submissionPayload = submissionSyncPayload` (or directly using `submissionSyncPayload` variable instead of `submissionPayload` result but maybe there is something I am missing. If that is the case, maybe we can remove this line and directly refer to `submissionSyncPayload` in the assertions below since `submissionSyncObservable` is created by the test code anyway. What do you think?

Reply:
I agree and applied. Assigning the subscribed value does not test actual logic (but only that the correct values are returned by the mock/predefined observable which is not needed by obvious reason). This is the case for both subscriptions.